Asterisk 10, 11, 13, 14 compatibility fixes. And a few other fixes.

wdoekes opened the pull request:
Here you go.

wdoekes committed:
Parse a +CMGS notification and hack it to send an event with reference.

Author: me, some time in the past.
Source: /~dexter/work/asterisk-chan-dongle/patch.patch.txt

wdoekes committed:
Add Asterisk 10 support.

Author: me, some time in the past.
Source: /~dexter/work/asterisk-chan-dongle/patch2.patch.txt

wdoekes committed:
Add more Asterisk 10 support.

Author: me, some time in the past.
Source: /~dexter/work/asterisk-chan-dongle/patch3.patch.txt

wdoekes committed:
Add Asterisk 11 support.

Author: me, in August 2013.

wdoekes committed:
Add .gitignore.

wdoekes committed:
Add Asterisk 13 support.

Author: novaktmp, 2014-12-01 17:42:52
Source: https://forum.openwrt.org/viewtopic.php?id=54343

Slight adaptations by me.

wdoekes committed:
Translate NUL to @ in incoming SMS.

Author: mio
Source: https://github.com/bg111/asterisk-chan-dongle/issues/153

Not using full 3GPP version from https://code.google.com/p/
  asterisk-chan-dongle/issues/detail?id=140 because of the issues
described there.

This should simply fix that you don't get a truncated SMS because of an
@ sign.

wdoekes committed:
Fix so Asterisk 1.8 compilation works again.

wdoekes committed:
Spaces cleanup. Asterisk version number cleanup.

bg111 commented:
i check ast_channel_cleanup() same as ast_channel_unref  but with NULL check.
So ast_channel_cleanup() is ok

yylyyl committed:
Applied chan-dongle-alphanumeric-pdu.patch.

USSD for E261 works now.

Source: https://code.google.com/p/asterisk-chan-dongle/issues/detail?id=17

wdoekes committed:
Move ASTERISK_VERSION_NUM setting to ./configure invocation:

Now do:

    ./configure --with-astversion=110102

if you have Asterisk version 11.1.2.

wdoekes committed:
Fix channel unlock issue in Asterisk >= 12.

Thanks @bg111 for the review.

wdoekes committed:
Clean up horrific ASTERISK_VERSION_NUM #if/#elif in channel_request.

wdoekes committed:
build: Properly force a recompile on config.h change.

Also move the ASTERISK_VERSION_NUM to config.h.

wdoekes committed:
Clean up format_cap_set stuff. Possibly fixing Local channel support.

thelan commented:
Hello,

Testes with asterisk 11.17.1. Works great!

mj84 commented:
Any change this will be merged with upstream?

wdoekes committed:
Clarify --with-astversion parameter in ./configure.

As suggested by @arekm. Note that "010813" does not work, as gcc will treat
it as base 8.

itamarjp commented:
please merge, thank you.

wdoekes committed:
Fix compile issues with master (Asterisk 14).

wdoekes committed:
Don't crash ongoing calls when receiving SMS.

According to the bug report by Marcelo Fernandes Vianna:
> Under some circumstances (if parsing the sms returns -1 in
> at_response.c), even if sms is disabled in dongle.conf,
> receiving a sms can still crash an ongoing call.

Thanks for the suggested patch. Refactored and merged.
Closes #7.

wdoekes committed:
Fix compile issues with Asterisk 14.

Declare AST_MODULE_SELF_SYM=__internal_$(module_name)_self.

wdoekes committed:
Fix decoding of SMS with non-language characters.

Character conversion assumed UCS-2, which handles only the Basic
Multilingual Plane, but various parties will have switched to using
UTF-16 there.

Fix suggested by @fadasi. Reported and tested by @pjmichel. Thanks!
Closes #8.

wdoekes committed:
Add macros for decoding TP-DA, TP-OA and TP-RA addresses.

wdoekes committed:
Support TON=5 with any NPI (instead of just 0) for alphanumeric address.

@fadasi observed addresses with 0xD0 and 0xD1 (NPI is UNKNOWN or E164)
in the wild. Fix comparisons to only check the TON.

Closes #9.

wdoekes committed:
Merge pull request #11 from bg111/master

Merge fixes from upstream bg111/asterisk-chan-dongle

wdoekes committed:
Merge pull request #10 from wdoekes/number_type_alnum_d0_d1

Allow D0 and D1 for addresses

Tested by @fadasi. Thanks!

wdoekes committed:
Fix so compilation with different include dirs work.

Now you can do this:

    ./configure --with-astversion=140000 \
        --with-asterisk=$HOME/src/asterisk-14/include
    make clean all

And:

    ./configure --with-astversion=10800 \
        --with-asterisk=$HOME/src/asterisk-1.8/include
    make clean all

Etc..

wdoekes committed:
Add travis file.

wdoekes committed:
Build fix for Asterisk 12.

wdoekes committed:
Add Travis badge. Update README to Markdown.

wdoekes committed:
Merge branch 'travis'

wdoekes committed:
Unbreak compiled module loading broken in c24a3150.

Fixes #15 but doesn't explain it yet.

wdoekes committed:
Better fix for breakage from c24a3150.

Closes #15.

fadasi committed:
Adds a DongleSendUSSD function (Device, USSD) dialplan function

Inspired from http://a.dmin.pro/?p=2712 (presumably October 4, 2011 by arturius)
Also adds DongleSendUSSD and DongleSendSMS dialplan functions examples

For more visibility and traceability, the unmodified code style has not been changed

wdoekes committed:
Merge pull request #16 from fadasi/master

Adds a DongleSendUSSD function (Device, USSD) dialplan function

Thanks @fadasi!

wdoekes committed:
Clarify ./bootstrap procedure. Drop ancient INSTALL docs.

Clarifies the "no makefile.am" problem from #18, but doesn't fix it.

waltercacau commented:
Any blockers on merging this PR?

wdoekes committed:
bootstrap: Fix file naming (not 'bootstap').

Reported by @k008 in ghwdoekes/asterisk-chan-dongle issue #20.

wdoekes committed:
build: Improve --with-astversion to take X.Y.Z version.

This should clarify a few things.

wdoekes committed:
build: Explain that you may use --with-asterisk on configure.

Closes ghwdoekes/asterisk-chan-dongle issue #20.

wdoekes committed:
build: Fix builds for Asterisk < 10.y.z. And fix Travis CI-test.

silviuchingaru commented:
I do not agree with you about chan_dongle destroying VoIP markets. The price of a gsm VoIP adapter on ebay is less than a dongle so I don't see were the harm is made by this module. 
Also think of how many markets Internet destroyed but also how many it created instead.

bg111 commented:
Used dongle can be buy at 5-10 usd, many operators selling new dongles for about 10 usd (below prime cost) for expanding to 3g service.

You seen any gateway for 10 usd ?

silviuchingaru commented:
No, I don't see any gateway at this price right now because they are relatively new products but in a year or 2 we will see for sure at same price 5 to 10 dolars , the used ones, on ebay. Dongles were also 50 dolars few years ago. 
I use chan dongle in a business enviroment on my own company not because they are ceap but because I like the integration of this module with Asterisk. Hope you will change your mind and release this module as a supported one, If you need, I can also donate for it. ;-)
Thank you for all your work by now!

bg111 commented:
I have line of products relative to this part of VoIP, i sell it and also offered as service. 
With existence of some money in pocket asking for donation -  my conscience not allow this :)

silviuchingaru commented:
:-)) Ok, understood. But I still pledge for a stable release of this module.

wdoekes committed:
pdu/tests: Clean up pdu_parse if-tree and restore test capability.

    make check

to build and run tests.

Cleanups were required to even begin attempting to fix #28.

wdoekes committed:
pdu/tests: Add tests for multipart 7bit encoded message decoding.

See bug #13. This creates stub tests. They're disabled, since it's
broken at the moment.

wdoekes committed:
char_conv: Handle 7bit encoded multi-part SMS correctly.

Most of this patch was taken from:
https://github.com/bg111/asterisk-chan-dongle/issues/183
==>
https://github.com/bg111/asterisk-chan-dongle/pull/214

Some changes were elided because:
- of uncertainty of their workings
- they did not improve the results of the available tests

----

From 2aba9dd30da7bfe69377c7b4cbf313ce01fa347c Mon Sep 17 00:00:00 2001
From: Rodrigo Freire <rfreire@rf.usersys.redhat.com>
Date: Fri, 28 Aug 2015 16:10:20 -0300
Subject: [PATCH] Resolves bg111/asterisk-chan-dongle#183

From: HPS <hsela...@gmail.com>
From: Shabbira <shabbira...@gmail.com>

Subject: Parses correctly multi-part SMS messages
Comitter: Rodrigo Freire <rfreire@rf.usersys.redhat.com>
Signed-off-by: Rodrigo Freire <rfreire@rf.usersys.redhat.com>

This patch was retrieved from
https://code.google.com/p/asterisk-chan-dongle/issues/detail?id=183
Now parses correctly multi-part SMS Messages. Verified on a Huawei
E303S.

----

Closes #13.

wdoekes committed:
pdu/parsing: Fix unsupported DCS value 0xF0.

Closes #28.

wdoekes committed:
docs: Add note about avoiding "SMS-injection".

Closes #24. Thanks Lynxie (@lnx13) for the report.

wdoekes committed:
logging: Add missing LFs to WARNINGs/ERRORs.

wdoekes committed:
travis: Make check after the build.

wdoekes committed:
decoding: Decode 7bit alphabet to unicode.

A better fix than the previous @ replacement in 897b27c27.

wdoekes committed:
cleanup: Fix a few typo's.

wdoekes committed:
cleanup: Fix some more typo's. The locks all seem okay.

wdoekes committed:
cleanup/fix: Undo pvt=NULL "fix" which was bad.

Fixes #30 I think.

wdoekes committed:
cleanup: Remove leftover round_robin_mtx comments.

It appears there previously was a global round_robin struct that had to
be mutex protected, but it has been refactored to use locally scoped vars
instead.

wdoekes committed:
cleanup: Unindent/flatten pvt_start() in chan_dongle.c.

wdoekes committed:
fix: Fix incidental deadlock with Jitterbuffer or Asterisk 12+.

Closes #19, reported by @vitasgul, @multijohn and Garrone Joseph
(@garronej).

Cause found through diligent bug reporting and testing by Garrone.

garronej committed:
adding instructions for applying Jitter buffer and automatic gain control

wdoekes committed:
Merge pull request #34 from garronej/master

Add instructions for applying Jitter buffer and automatic gain control

Thanks @garronej!

wdoekes committed:
Ignore """+CMTI: "SM",-1""" with negative index.

Instead of disconnecting a standing call, we ignore the message.

Closes #35. Requested by @vitasgul.

wdoekes committed:
Attempt to open the dongle in exclusive mode using TIOCEXCL.

Closes #36. Requested by @Andrik45719.

micmac1 committed:
Fix audio endianess problem on big endian systems.

This fixes broken audio from asterisk toward the GSM network. Issue was
originally reported here: https://github.com/openwrt/telephony/issues/7.
Github user @ljakob provided this patch and reports success:

(Sep 9, 2015)
> Hi, I've used the patch over the last month on my box and it works
> perfectly. Audio gain from my old x86 installation had to be copied but
> that was expected. There have been at least 100 calls without any
> problems.

Signed-off-by: Sebastian Kemper <sebastian_ml@gmx.net>
Code cleaned up by wdoekes.

Closes #42.

wdoekes committed:
Add Asterisk 14/15 Travis checks.

wdoekes committed:
Revert "fix Sparc loud noise problem #201"

This reverts commit ab939532c71b991de2f66582d7ee2c553b6a918e.

Apparently there was too much byteswapping (on big-endian machines)
going on with both ab93953 and 53e83a4. Reverting ab93953 appears to
work for @Infactum.

Closes #44.

hselasky committed:
Fix for stuck SMS.

It has been observed that CMTI notifications can stop when the internal
memory of the E1750 dongle is full. Instead an SMMEMFULL indication is received.

This patch includes multiple fixes:
- Always poll SMS messages during startup.
- Only poll one SMS message at a time, by keeping a bitmap of polled messages.
- Only delete message after successful reception.

Assume that no more than 256 SMS's will be stored by the external SIM card.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hapylestat commented:
@bg111 lol, "VoIP market" is not something, what usually ppl needed.  Default usage scenario for anyone in the world: mobila data plan + viber, skype, etc. 

Purely, VOIP tech is used by companies, who need to manage their internal communications with possibility to connect to external phone network.  Any normal company will not have any deal with schoolboy as he basically will have not permission/license for such type of activity, which is required by regulations.

Think that market depends on ppl, who using VOIP - is a schoolboy way of thinging

wdoekes committed:
Lower default timeout of 2sec to 5sec and 15sec to 40sec

This should probably fix call drop issues with E3131A modem (same as
E303).

I cannot quickly find documentation about what the timeout "should" be,
but using 3 timeout classes with short/medium/long makes more sense to
me than having the classes defined by the duration.

Please report any issues that arise from this change.

Suggested-by: @mthqwork
Closes: #50

wdoekes committed:
Improve asterisk/modules dir autodetection

Previously ./configure would continue after not finding DESTDIR for the
module and then fail at 'make install'. Now it requires you to set
DESTDIR if autodetection fails. Also added lib64 in the default paths to
try.

Reported-by: @alaa2003
Closes: #49

wdoekes committed:
Hide -Wimplicit-fallthrough= false positives

gcc apparently groks "/* fall through */" as well as __attribute__
((fallthrough)) which probably is not supported on all compilers.

wdoekes committed:
pdu/parsing: Fix unsupported DCS value 19

After looking at specs from
http://www.3gpp.org/ftp//Specs/archive/23_series/23.038/ it looks like
it should allow more than we did.

Closes #41 (I think).

wdoekes committed:
pdu/parsing: Allow all values for TP-PID

Closes #53, reported by @standardgbg.

See: http://www.3gpp.org/ftp//Specs/archive/23_series/23.040/23040-e00.zip

wdoekes committed:
travis: Unbreak CI

grray committed:
USSD encoding fix for E3531

Huawei E3531 works fine with firmware 21.318.29.00.769, voice and sms, except USSD.
Adding it to seven_bit_modems[] solves issue. (#56)

miopa committed:
send sms to local and internal network numbers

shetubd commented on README.md:
```diff
@@ -62,6 +62,18 @@ Here is an example for the dialplan:
     exten => _X.,1,Dial(Dongle/r1/${EXTEN})
     exten => _X.,n,Hangup
 
+    exten => *#123#,1,DongleSendUSSD(dongle0,${EXTEN})
```
I got this message for sending USSD from xlite sip client. From astrisk cli ussd send without problem.

[2018-08-17 23:58:26] WARNING[12618][C-00000001]: pbx.c:2866 pbx_extension_helper: No application 'DongleSendUSSD' for extension (dongle-trunk, *2#, 1)
  == Spawn extension (dongle-trunk, *2#, 1) exited non-zero on 'SIP/1007-00000001'

garronej committed:
Better params for the JITTERBUFFER in README

misuzu committed:
Add with-iconv

wdoekes committed:
Trim trailing garbage from provider name causing ast_json_vpack error

The AT+COPS command has been observed to return similar but different
provider names. Sometimes with a trailing illegal character, or a
trailing @.

Dongle make/model E1550 (patched for voice support), identified by Linux
as "Huawei Technologies Co., Ltd. E161/E169/E620/E800 HSDPA Modem".

Example output:

    AT+COPS?
    +COPS: 0,2,"25020",2

    AT+COPS=?
    +COPS: (2,"TELE2 RU","TELE2 RU","25020",2),
           (1,"MTS-RUS","MTS","25001",2),
           (1,"Beeline","Beeline","25099",2),
           (1,"MegaFon)

    AT+COPS=?
    +COPS: (2,"TELE2 RU","TELE2 RU","25020",2),
           (1,"MegaFon RUS","MegaFon","25002",2),
           (1,"Beeline","Beeline","25099",2),
           (1,)

    AT+COPS=?
    +COPS: (2,"Tele2@","Tele2�","25020",2),
           (1,"MTS-RUS","MTS","25001",2),
           (1,"Beeline","Beeline","25099",2),
           (1,"MegaFon RUS)

(that replacement character (U+FFFD) is probably something else; likely a
single octet in the >=128 range)

A dongle change (same make/model) did not help. A simcard change did not
help. Likely culprit for now seems to be the provider (Russian Tele2?).

This changeset drops trailing "@" and trailing non-printables from the
provider name. It would still switch from "TELE2 RU" to "Tele2", but not
to something with unreadable characters.

Closes #39.

This also fixes this error:

    json.c:607 ast_json_vpack: Error building JSON from '{s: s, s: s}':
      Invalid UTF-8 string.

Generated at:

    for (idx = 0; idx < ITEMS_OF(dev_vars); ++idx) {
        # dev_vars[1] == { "DONGLEPROVIDER", pvt->provider_name }
        pbx_builtin_setvar_helper(channel, dev_vars[idx].name,
                                  dev_vars[idx].value);
    }

Because newer Asterisken use JSON to publish these messages, and that
character (which isn't U+FFFD) is illegal UTF-8.

Closes #69.

Thanks go to sergio (@532910) for debugging and testing this issue.

wdoekes committed:
Fix missing trailing LF in README

wdoekes committed:
Add chanvar setting ast_debug to aid in future debugging

wdoekes committed:
travis: Alter asterisk include files remote path

wdoekes committed:
Merge pull request #72 from misuzu/with-iconv

Add --with-iconv option to specify a different path

wdoekes committed:
Disable Asterisk 1.8 travis test

It's old, and conflicts with #75

mvonbuelow committed:
Disable travis test with unstable asterisk versions

mvonbuelow committed:
Added complete GSM-7 tables; Implemented GSM-7 escape

mvonbuelow committed:
Implemented concatenated SMS support

magcks committed:
Merge branch 'master' into master

wdoekes committed:
Merge pull request #75 from magcks/master

Concatenated SMS support; Complete locking and single shift tables.

Thanks Max von Buelow!

wdoekes committed:
Rename enque to enqueue, and remove fishy function pointer casts

The casting of functions taking fewer parameters than were given
probably did work on most platforms with common calling conventions. But
the calling convention is not defined by us. And the C-style cast
removes all other sanity checks as well.

mvonbuelow committed:
Seperated PDU parsing (and building) into hex parsing, actual PDU decoding and UCS-2 decoding; Implemented delivery reports; Fixed USSD generating

hselasky committed:
Fix bug in SMS parsing logic.
Try to use length field at beginning of TP-DA, instead of assuming fixed length.

[dongle0] Error parsing incoming message '+CMGR: 0,,24
06917429000102068E0A917419778290023020020200400230200202004000' at position 70:
Good report, but delivery failed

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

mvonbuelow committed:
E3531 support

wdoekes committed:
Merge pull request #80 from magcks/E3531s2

"[ignore] AT+CNUM errors [returned by] E3531s-2"

Thanks @magcks!

wdoekes committed:
Add configure.ac checks for sqlite3

Closes #74, reported by @ruijorgesilva1.

wdoekes committed:
Merge pull request #79 from hselasky/master

Thanks @hselasky.

mvonbuelow committed:
Resolved conflicts w/ #79

wdoekes committed:
Merge pull request #77 from magcks/unhexpdu

Reworked PDU parsing; Delivery reports

mvonbuelow committed:
Fixed issue #85

wdoekes committed:
Merge pull request #86 from magcks/unhexpdu

Fixed issue #85

hselasky committed:
Use incoming_sms_index field as indication of incoming SMS instead of incoming_sms.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix compile warning. Ensure enum2str() is declared.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix uninitialized use of res variable.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix compile warnings.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Remove unused variables.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix compile warning for clang: Unsequenced modification of i.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix variable type.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Always receive next SMS regardless of error.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix integer to pointer conversion.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

hselasky committed:
Fix regression issue after PDU parsing was separated.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

wdoekes committed:
Merge pull request #90 from hselasky/master

Fixes for regression issues and some improvements.

Thanks Hans Petter Selasky!

Commit:
Removed some magic. Sorry, I'm not a wizard

phcoder committed:
Add guard against NULL args.payload

NULL args.payload causes a crash in the call to strlen(args.payload)
while computing arguments for send_sms.

miopa committed:
Use Asterisk function `ast_unescape_c` on SMS text when sending via AMI

This change was warranted because AMI is a text based protocol and the newline can't be included in the payload.
Standard backslash escape sequences are used, e.g. '\n' for newline and '\t' for tabulator characters.

wdoekes committed:
Merge pull request #101 from miopa/patch-2

Use Asterisk function `ast_unescape_c` on SMS text when sending via AMI

Thanks Damjan Janevski!

wdoekes committed:
Merge pull request #99 from phcoder/payload

Add guard against NULL args.payload

Thanks Vladimir Serbinenko!

Commit:
fix ucs-2 empty sms bug

wdoekes committed:
Merge pull request #96 from p-velasco/pin-implementation

"Removed some magic. Sorry, I'm not a wizard"

Thanks @p-velasco!

Commit:
Better type safe sms inbox

Commit:
Suppress errors on initial sms poll

wdoekes committed:
Merge pull request #107 from 4e4o/master

Better type safety of SMS handling and error suppression on initial SMS poll.

Thanks @4e4o

h31p committed:
FreeBSD has slightly another place for lock files

hardillb committed:
A better change to the logging

mghadam committed:
Updated interfaces for E171

The correct device interfaces for HUAWEI E171 modem is  3,2  . the 1,2 interface order does not work. 

Model: E171
Firmware: 21.156.00.00.143
IMEI: 861496015775578

mghadam committed:
Added E153Du-1 device id ports

Updated ports for E153Du-1
IMEI: 868672001033577

129tyc committed:
fix Cannot parse UCS-2 error (#131)

Thanks @129tyc!

wdoekes committed:
Abort compilation on warnings

wdoekes committed:
Fix compiler warnings; add casts, add const when easy, remove unused

wdoekes committed:
Try GitHub Actions instead of "Free Plan Will Always Be Free"-Travis

wdoekes committed:
Fix make check compiler warnings and check errors

msg had been partially removed in 868bbd4.

wdoekes committed:
Merge branch 'fix-compiler-warnings'

wdoekes committed:
Better variable locality in at_enqueue_initialization

(No random "cmd12" constants.)

wdoekes committed:
Add Quectel support

Totally untested. Might close #144.

Based on this patch:

    From d332203e6fb869ee9f7f48cd6fd794394888d284 Mon Sep 17 00:00:00 2001
    From: Sebastian Kemper <sebastian_ml@gmx.net>
    Date: Thu, 9 Sep 2021 11:14:30 +0200
    Subject: [PATCH] Dongle to Quectel changes

    Signed-off-by: Sebastian Kemper <sebastian_ml@gmx.net>
    ---
     at_command.c | 4 ++--
     pdiscovery.c | 1 +
     2 files changed, 3 insertions(+), 2 deletions(-)

    diff --git a/at_command.c b/at_command.c
    index 4882d82..684e821 100644
    --- a/at_command.c
    +++ b/at_command.c
    @@ -33,7 +33,7 @@ static const char cmd_at[] 	 = "AT\r";
     static const char cmd_chld1x[]   = "AT+CHLD=1%d\r";
     static const char cmd_chld2[]    = "AT+CHLD=2\r";
     static const char cmd_clcc[]     = "AT+CLCC\r";
    -static const char cmd_ddsetex2[] = "AT^DDSETEX=2\r";
    +static const char cmd_ddsetex2[] = "AT+QPCMV=1,0\r";

     /*!
      * \brief Format and fill generic command
    @@ -128,7 +128,7 @@ EXPORT_DEF int at_enqueue_initialization(struct cpvt *cpvt, at_cmd_t from_comman
     	static const char cmd15[] = "AT+CREG?\r";
     	static const char cmd16[] = "AT+CNUM\r";

    -	static const char cmd17[] = "AT^CVOICE?\r";
    +	static const char cmd17[] = "AT+QPCMV?\r";
     //	static const char cmd18[] = "AT+CLIP=0\r";
     	static const char cmd19[] = "AT+CSSN=1,1\r";
     	static const char cmd20[] = "AT+CMGF=0\r";
    diff --git a/pdiscovery.c b/pdiscovery.c
    index 006c5d8..c599c23 100644
    --- a/pdiscovery.c
    +++ b/pdiscovery.c
    @@ -68,6 +68,7 @@ static const struct pdiscovery_device device_ids[] = {
             { 0x12d1, 0x14ac, { 4, 3, /* 0 */ } },          /* E153Du-1 : thanks mghadam */
     	{ 0x12d1, 0x1436, { 4, 3, /* 0 */ } },		/* E1750 */
     	{ 0x12d1, 0x1506, { 3, 2, /* 0 */ } },		/* E171 firmware 21.x : thanks Sergey Ivanov */
    +	{ 0x2c7c, 0x0125, { 1, 4, /* 0 */ } },		/* Dongle EC25-A LTE modem */
     };

     static struct discovery_cache cache;

wdoekes committed:
Quick fix for mips-openwrt-linux-musl-gcc maybe-uninitialized warning

(Definite false positive.)

wdoekes committed:
Don't do -Werror by default, use ./configure --enable-warnings-as-errors

rojer committed:
Remove use_ucs2_encoding (#146)

It's no longer used

wdoekes committed:
Merge branch 'add-quectel-support'

Non-Quectel tested by Deomid Ryabkov. Also includes better -Werror
behaviour.

Quectel support untested thusfar. We'll see how that goes.

rojer committed:
Fix GSM7 decoding on big endian targets (#145)

The uint16_t bytes are the other way around.

wdoekes committed:
Do not choke on failed AT^CVOICE? or AT+QPCMV? during startup

ATQ_CMD_DECLARE_STI adds ATQ_CMD_FLAG_IGNORE so the error response on
either command does not abort the entire initialization task.

Without it, we migh get:

    [AT+QPCMV?]
    remove command 'AT+QPCMV' expected response 'OK' real 'ERROR' cmd
      16/22 flags 0x01 from queue
    remove task with 22 command(s) begin with 'AT' expected response
      'OK' from queue

where the failed AT+QPCMV? aborts the 22-item init task.

Closes #148, reported and tested by @AlexHighTower.

wdoekes committed:
Remove probably-useless code at voice init failure

> i think these lines can also be removed, since ATQ_CMD_DECLARE_STI
> will plow ahead anyway, no need to manually restart the sequence.

Dixit, @rojer in #148. I don't know. Sounds plausible.

wdoekes committed:
Add autoheader to ./bootstrap; add HAVE_LIBSQLITE3; error on smsdb init fail

wdoekes committed:
Add PRI_time_t in case we want to printf time_t values

wdoekes committed:
Fix formatting in error.h

wdoekes committed:
Revert "Updated interfaces for E171"

This reverts commit f1517b427f7542f278ba6b8e504c5debb563bd9a.

See discussion on:
https://github.com/wdoekes/asterisk-chan-dongle/commit/f1517b427f7542f278ba6b8e504c5debb563bd9a

There @Vald3m argues that the old values were correct.

micmac1 committed:
Consider long long (lld) for time_t (#173)

Some libc implementations are shifting toward using 64-bit time_t for
all arches to solve the year 2038 problem. musl is one of them, see [1].

So when compiling for a 32-bit arch time_t will be long long. This
commit adds this option.

[1] https://musl.libc.org/time64.html

Signed-off-by: Sebastian Kemper <sebastian_ml@gmx.net>

Commit:
Added udhl -= 1; (= udhl -= iei_len) after reading the data byte in both case 0x24 and case 0x25. Also corrected the wrong comment on 0x25 (it's "Locking Shift", not "Single Shift").

wdoekes committed:
Merge pull request #183 from giraygokirmak/master

SMS received containing Turkish chars gets scrambled

Thanks!
